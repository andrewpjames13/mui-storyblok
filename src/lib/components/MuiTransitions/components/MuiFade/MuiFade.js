import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { Fade } from '@material-ui/core';
import StoryBlok from '../../../../utils/Storyblok';
import MuiGridList from '../../../MuiGridList/MuiGridList';

const MuiFade = ({
  rootClass,
  content,
  fadeIn,
  timeout,
}) => {
  const components = {
    MuiGridList,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <div className={styles.root}>
      <Fade
        in={fadeIn}
        timeout={timeout}
      >
        <div>
          {content.map((item, index) => createElement(
            components[item.component],
            Object.assign(item, { key: index }),
          ))}
        </div>
      </Fade>
    </div>
  );
};

export default MuiFade;

MuiFade.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: bool
   * If true, the component will transition in.
   */
  fadeIn: PropTypes.bool,
  /**
   * mui props: number | string
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * Set to 'auto' to automatically calculate transition time based on height.
   */
  timeout: PropTypes.string,
  /**
   * Content passed to render
   * components: MuiGridList
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiFade.defaultProps = {
  fadeIn: true,
  timeout: 'auto',
  rootClass: [],
};
