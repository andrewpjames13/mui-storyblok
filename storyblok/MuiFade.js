module.exports = {
  name: 'MuiFade',
  display_name: null,
  updated_at: '2020-05-13T01:35:51.733Z',
  schema: {
    rootClass: {
      type: 'options',
      description: '* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component.',
    },
    content: {
      type: 'bloks',
      default_value: '',
      restrict_components: true,
      component_whitelist: [
        'Blok',
      ],
      maximum: '1',
      required: true,
      description: '',
    },
    fadeIn: {
      type: 'number',
      description: 'Number of milliseconds for component to transition in.',
      default_value: '300',
    },
    enter: {
      type: 'number',
      description: 'Number of milliseconds for component transition to effect on screen.',
      display_name: '',
      default_value: '1500',
    },
    exit: {
      type: 'number',
      description: 'Number of milliseconds for component transition to effect off screen.',
      display_name: '',
      default_value: '1500',
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiFade',
  component_group_uuid: null,
};
