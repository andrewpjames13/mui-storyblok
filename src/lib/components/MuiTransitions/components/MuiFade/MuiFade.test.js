import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiFade from './MuiFade';

function setup() {
  const props = {
    enter: 1500,
    exit: 1500,
    content: [{
      component: 'Blok',
      content: [{
        component: 'MuiGrid',
        content: [{
          component: 'MuiGridItem',
          content: [{
            component: 'MuiCard',
            content: [{
              component: 'MuiCardContent',
              content: [{
                component: 'MuiTypography',
                content: [{
                  component: 'MuiText',
                  text: 'Juice Box',
                }],
              }],
            }],
          }],
        }],
      }],
    }],
  };

  const comp = shallow(<MuiFade {...props} />);
  return { comp, props };
}

describe('<MuiFade />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiFade {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiGridList', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });

  it('should change state after transitionIn time is ran', () => {

    // const { comp } = setup();
    // expect(comp.find('WithStyles(ForwardRef(Fade))').first().props().in).toEqual(false);
    // setTimeout(() => {
    //   expect(comp.find('WithStyles(ForwardRef(Fade))').first().props().in).toEqual(true);
    // }, 1000);
  });
});
