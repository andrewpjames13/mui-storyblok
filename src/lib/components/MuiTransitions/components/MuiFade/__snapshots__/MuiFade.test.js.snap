// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiFade /> should render MuiGridList 1`] = `ShallowWrapper {}`;

exports[`<MuiFade /> snapshot 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  />
</div>
`;
